src/mbgl/sprite/sprite_atlas.cpp

```diff
@@ -317,73 +317,34 @@ void SpriteAtlas::updateDirty() {
     }
 
-    if (!texture) {
-        texture = context.createTexture();
-        context.activeTexture = unit;
-        context.texture[unit] = *texture;
 #if not MBGL_USE_GLES2
-        MBGL_CHECK_ERROR(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAX_LEVEL, 0));
-#endif // MBGL_USE_GLES2
-        // We are using clamp to edge here since OpenGL ES doesn't allow GL_REPEAT on NPOT textures.
-        // We use those when the pixelRatio isn't a power of two, e.g. on iPhone 6 Plus.
-        MBGL_CHECK_ERROR(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE));
-        MBGL_CHECK_ERROR(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE));
-        fullUploadRequired = true;
-    } else if (context.texture[unit] != *texture) {
-        context.activeTexture = unit;
-        context.texture[unit] = *texture;
-    }
+//    const bool wasDirty = dirtyFlag;
+#endif
 
-    GLuint filter_val = linear ? GL_LINEAR : GL_NEAREST;
-    if (filter_val != filter) {
-        context.activeTexture = unit;
-        MBGL_CHECK_ERROR(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, filter_val));
-        MBGL_CHECK_ERROR(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, filter_val));
-        filter = filter_val;
+    if (!texture) {
+        texture = context.createTexture(image, unit);
+        dirtyFlag = false;
     }
 
     if (dirtyFlag) {
-        std::lock_guard<std::recursive_mutex> lock(mtx);
-
-        context.activeTexture = unit;
-        if (fullUploadRequired) {
-            MBGL_CHECK_ERROR(glTexImage2D(
-                GL_TEXTURE_2D, // GLenum target
-                0, // GLint level
-                GL_RGBA, // GLint internalformat
-                pixelWidth, // GLsizei width
-                pixelHeight, // GLsizei height
-                0, // GLint border
-                GL_RGBA, // GLenum format
-                GL_UNSIGNED_BYTE, // GLenum type
-                data.get() // const GLvoid * data
-            ));
-            fullUploadRequired = false;
-        } else {
-            MBGL_CHECK_ERROR(glTexSubImage2D(
-                GL_TEXTURE_2D, // GLenum target
-                0, // GLint level
-                0, // GLint xoffset
-                0, // GLint yoffset
-                pixelWidth, // GLsizei width
-                pixelHeight, // GLsizei height
-                GL_RGBA, // GLenum format
-                GL_UNSIGNED_BYTE, // GLenum type
-                data.get() // const GLvoid *pixels
-            ));
-        }
-
+        context.updateTexture(*texture, image, unit);
         dirtyFlag = false;
+    }
+
+    context.bindTexture(*texture, unit,
+                        linear ? gl::TextureFilter::Linear : gl::TextureFilter::Nearest);
 
 #if not MBGL_USE_GLES2
-        // platform::showColorDebugImage("Sprite Atlas", reinterpret_cast<const char*>(data.get()),
-        //                               pixelWidth, pixelHeight, pixelWidth, pixelHeight);
+//    if (wasDirty) {
```

Comment:
Yeah; that code is there for debugging purposes with GLFW. If you need to visualize the glyph atlas, you can comment it out. That code used to be there before as well, I just rewrote it to match the new texture management.